packages/flutter/lib/src/cupertino/sheet.dart

```diff
@@ -0,0 +1,767 @@
+    Animation<double> animation,
+    Animation<double> secondaryAnimation,
+    Widget child,
+  ) {
+    final bool linearTransition = route.popGestureInProgress;
+    return CupertinoSheetTransition(
+      primaryRouteAnimation: animation,
+      secondaryRouteAnimation: secondaryAnimation,
+      linearTransition: linearTransition,
+      child: _CupertinoDownGestureDetector<T>(
+        enabledCallback: () => true,
+        onStartPopGesture: () => _startPopGesture<T>(route),
+        child: child,
+      ),
+    );
+  }
+
+  @override
+  bool canTransitionTo(TransitionRoute<dynamic> nextRoute) {
+    return nextRoute is _CupertinoSheetRouteTransitionMixin;
+  }
+
+  @override
+  Widget buildTransitions(
+    BuildContext context,
+    Animation<double> animation,
+    Animation<double> secondaryAnimation,
+    Widget child,
+  ) {
+    return buildPageTransitions<T>(this, context, animation, secondaryAnimation, child);
+  }
+}
+
+class _CupertinoDownGestureDetector<T> extends StatefulWidget {
+  const _CupertinoDownGestureDetector({
+    super.key,
+    required this.enabledCallback,
+    required this.onStartPopGesture,
+    required this.child,
+  });
+
+  final Widget child;
+
+  final ValueGetter<bool> enabledCallback;
+
+  final ValueGetter<_CupertinoDownGestureController<T>> onStartPopGesture;
+
+  @override
+  _CupertinoDownGestureDetectorState<T> createState() => _CupertinoDownGestureDetectorState<T>();
+}
+
+class _CupertinoDownGestureDetectorState<T> extends State<_CupertinoDownGestureDetector<T>> {
+  _CupertinoDownGestureController<T>? _downGestureController;
+
+  late VerticalDragGestureRecognizer _recognizer;
+
+  @override
+  void initState() {
+    super.initState();
+    _recognizer =
+        VerticalDragGestureRecognizer(debugOwner: this)
+          ..onStart = _handleDragStart
+          ..onUpdate = _handleDragUpdate
+          ..onEnd = _handleDragEnd
+          ..onCancel = _handleDragCancel;
+  }
+
+  @override
+  void dispose() {
+    _recognizer.dispose();
+
+    // If this is disposed during a drag, call navigator.didStopUserGesture.
+    if (_downGestureController != null) {
+      WidgetsBinding.instance.addPostFrameCallback((_) {
+        if (_downGestureController?.navigator.mounted ?? false) {
+          _downGestureController?.navigator.didStopUserGesture();
+        }
+        _downGestureController = null;
+      });
+    }
+    super.dispose();
+  }
+
+  void _handleDragStart(DragStartDetails details) {
+    assert(mounted);
+    assert(_downGestureController == null);
+    _downGestureController = widget.onStartPopGesture();
+  }
+
+  void _handleDragUpdate(DragUpdateDetails details) {
+    assert(mounted);
+    assert(_downGestureController != null);
+    _downGestureController!.dragUpdate(
+      // Devide by size of the sheet. The gap between the top of the sheet and
+      // top of the screen is 0.08.
+      details.primaryDelta! / (context.size!.height - (context.size!.height * 0.08)),
```

Comment:
Nit comment from the other PR about this 0.08 value: https://github.com/flutter/flutter/pull/161324/files#r1911057792